src/Buoyancy/teos_10.jl

```diff
@@ -0,0 +1,284 @@
+       ((R₃₃₀ * s + R₂₃₀) * s + R₁₃₀) * s + R₀₃₀) * τ +
+      (((R₄₂₀ * s + R₃₂₀) * s + R₂₂₀) * s + R₁₂₀) * s + R₀₂₀) * τ +
+     ((((R₅₁₀ * s + R₄₁₀) * s + R₃₁₀) * s + R₂₁₀) * s + R₁₁₀) * s + R₀₁₀) * τ +
+    (((((R₆₀₀ * s + R₅₀₀) * s + R₄₀₀) * s + R₃₀₀) * s + R₂₀₀) * s + R₁₀₀) * s + R₀₀₀
+
+@inline r′(τ, s, ζ) = ((r′₃(τ, s) * ζ + r′₂(τ, s)) * ζ + r′₁(τ, s)) * ζ + r′₀(τ, s)
+
+#####
+##### Full density
+#####
+
+"""
+    ρ(Θ, S, p)
+
+Returns the in-situ density of seawater with state (Θ, S, p) using the 55-term
+polynomial approximation TEOS-10 described in Roquet et al. (§3.1, 2014).
+
+# Inputs
+    `Θ`: conservative temperature (ITS-90) [°C]
+    `S`: absolute salinity [g/kg]
+    `p`: sea pressure [dbar] (i.e. absolute pressure - 10.1325 dbar)
+
+# Output
+    `ρ`: in-situ density [kg/m³]
+
+# References
+Roquet, F., Madec, G., McDougall, T. J., Barker, P. M., 2014: Accurate polynomial
+    expressions for the density and specific volume of seawater using the TEOS-10
+    standard. Ocean Modelling.
+"""
+@inline ρ(Θ, S, p) = _ρ(τ(Θ), s(S), ζ(p))
+
+@inline _ρ(τ, s, ζ) = r₀(ζ) + r′(τ, s, ζ)
+
+#####
+##### Thermal expansion fit
+#####
+
+const α₀₀₀ = -6.5025362670e-01
+const α₁₀₀ =  1.6320471316e+00
+const α₂₀₀ = -2.0442606277e+00
+const α₃₀₀ =  1.4222011580e+00
+const α₄₀₀ = -4.4204535284e-01
+const α₅₀₀ =  4.7983755487e-02
+const α₀₁₀ =  1.8537085209e+00
+const α₁₁₀ = -3.0774129064e+00
+const α₂₁₀ =  3.0181275751e+00
+const α₃₁₀ = -1.4565010626e+00
+const α₄₁₀ =  2.7361846370e-01
+const α₀₂₀ = -1.6246342147e+00
+const α₁₂₀ =  2.5086831352e+00
+const α₂₂₀ = -1.4787808849e+00
+const α₃₂₀ =  2.3807209899e-01
+const α₀₃₀ =  8.3627885467e-01
+const α₁₃₀ = -1.1311538584e+00
+const α₂₃₀ =  5.3563304045e-01
+const α₀₄₀ = -6.7560904739e-02
+const α₁₄₀ = -6.0212475204e-02
+const α₀₅₀ =  2.8625353333e-02
+const α₀₀₁ =  3.3340752782e-01
+const α₁₀₁ =  1.1217528644e-01
+const α₂₀₁ = -1.2510649515e-01
+const α₃₀₁ =  1.6349760916e-02
+const α₀₁₁ = -3.3540239802e-01
+const α₁₁₁ = -1.7531540640e-01
+const α₂₁₁ =  9.3976864981e-02
+const α₀₂₁ =  1.8487252150e-01
+const α₁₂₁ =  4.1307825959e-02
+const α₀₃₁ = -5.5927935970e-02
+const α₀₀₂ = -5.1410778748e-02
+const α₁₀₂ =  5.3278413794e-03
+const α₀₁₂ =  6.2099915132e-02
+const α₀₀₃ = -9.4924551138e-03
+
+"""
+    α(Θ, S, p)
+
+Returns the Boussinesq thermal expansion coefficient -∂ρ/∂Θ [kg/m³/K] computed using
+the 55-term polynomial approximation to TEOS-10 described in Roquet et al. (§3.1, 2014).
+
+# Inputs
+    `Θ`: conservative temperature (ITS-90) [°C]
+    `S`: absolute salinity [g/kg]
+    `p`: sea pressure [dbar] (i.e. absolute pressure - 10.1325 dbar)
+
+# Output
+    `α`: Boussinesq thermal expansion coefficient -∂ρ/∂Θ [kg/m³/K]
+
+# References
+Roquet, F., Madec, G., McDougall, T. J., Barker, P. M., 2014: Accurate polynomial
+    expressions for the density and specific volume of seawater using the TEOS-10
+    standard. Ocean Modelling.
+"""
+@inline α(Θ, S, p) = _α(τ(Θ), s(S), ζ(p))
```

Comment:
I was under the impression that [Young (2010)](https://journals.ametsoc.org/doi/full/10.1175/2009JPO4294.1) shows that we need to use depth rather than pressure in the EOS to get energy conservation in Boussinesq... so I'm surprised that TEOS-10 uses pressure. I guess my only worry is that by writing functions in terms of pressure, we give the false impression that we should insert the full non hydrostatic pressure into them, when in fact that procedure would violate conservation of energy (as shown by Young (2010)).

Is it okay to write these functions in terms of `p`, when we won't actually use the full `p`?